src/stores/contractStore.ts

```diff
@@ -0,0 +1,275 @@
+import { defineStore } from "pinia";
+import { Contract, toAe } from "@aeternity/aepp-sdk";
+import { Ref, ref } from "vue";
+import { useSdkStore } from "./sdkStore";
+import {
+  argsStringToArgs,
+  defaultCallOptions,
+  exampleContractCode,
+  getContract,
+  persistContract,
+  Result,
+} from "../utils/utils";
+import "../utils/toJsonExtensions";
+
+export const useContractStore = defineStore("contract", () => {
+  const compileData: Ref<{ contractCode: string }> = ref({
+    contractCode: structuredClone(exampleContractCode),
+  });
+  const compileResult: Ref<Result<{ aci: string; byteCode?: string }>> = ref(
+    new Result()
+  );
+
+  const deployData: Ref<{
+    args: string;
+    options: {
+      amount: number;
+      callData: string;
+      fee: null;
+      gas: null;
+      gasPrice: number;
+    };
+  }> = ref({
+    args: "",
+    options: structuredClone(defaultCallOptions),
+  });
+  const deployResult: Ref<Result<string>> = ref(new Result());
+
+  const callStaticData: Ref<{ args: string; func: string; gas: number }> = ref({
+    func: "example",
+    gas: 1000000,
+    args: "",
+  });
+  const callStaticResult: Ref<Result<string>> = ref(new Result());
+
+  const callData: Ref<{
+    args: string;
+    func: string;
+    options: {
+      amount: number;
+      callData: string;
+      fee: null;
+      gas: null;
+      gasPrice: number;
+    };
+  }> = ref({
+    func: "example",
+    args: "",
+    options: structuredClone(defaultCallOptions),
+  });
+  const callResult: Ref<Result<string>> = ref(new Result());
+
+  let contractInstance: Contract<any> | undefined = undefined;
+
+  const sdkStore = useSdkStore();
+
+  async function compileContractFromSource() {
+    compileResult.value.setInfo("Compiling contract source");
+    resetDeployAndCallData();
+
+    await sdkStore.aeSdk?.compilerApi
+      .compileBySourceCode(compileData.value.contractCode)
+      .then((result) => {
+        compileResult.value.setFinal("Compiled from source", {
+          aci: JSON.stringify(result?.aci, null, 2),
+          byteCode: result.bytecode,
+        });
+
+        persistContract(
+          compileData.value.contractCode,
+          compileResult.value.data?.aci || "",
+          deployResult.value.data
+        );
+      })
+      .catch((error) => {
+        if (error instanceof Error)
+          compileResult.value.setError(
+            error.message.replaceAll("compile error:\n", "")
+          );
+        return undefined;
+      });
+  }
+
+  async function initializeContractFromAci(
+    contractAddress: string,
+    aci: string
+  ) {
+    resetDeployAndCallData();
+    deployResult.value.setInfo("Instantiating Contract at address ...");
+
+    await sdkStore.aeSdk
+      ?.initializeContract({
+        ...{ aci: JSON.parse(aci) },
+        address: `ct_${contractAddress.replace("ct_", "")}`,
```

Comment:
what does conversion do? removing something just to add it back in the same place? or are you unsure about the format you receive the address in?

Reply:
was an issue in the sdk beta, seems to be fixed now